### ISSUE TITLE
Rethink the binary distribution process

### ISSUE THREAD

**Issue by FelixKratz** [FelixKratz/SbarLua#15]
As soon as the v1.0 milestone is reached, this project will have a stable release system. The release binary of the sketchybar module needs to be distributed in a properly managed way with proper updating etc. I am not sure if distribution through brew is the proper way to do it for this binary.

**Comment by khaneliman** [comment]
I know I'm going to add it to nixpkgs, when I figure out how to do a lua pkg on there. But, I think from what I was reading people used something like luarocks?

**Comment by shajra** [comment]
@khaneliman I have SbarLua building with Nix and have integrated it into my setup.   The packaging of the shared library isn't so bad:

https://github.com/shajra/shajra-provisioning/blob/main/build/nixpkgs/packages/sketchybar-lua/default.nix

And the the way I integrate it with the Lua configuration is more-or-less how @FelixKrat illustrates in his examples:

- [template](https://github.com/shajra/shajra-provisioning/blob/main/home/modules/base/gui/darwin/xdg/configFile/sketchybar/sketchybarrc)
- [substitution](https://github.com/shajra/shajra-provisioning/blob/main/home/modules/base/gui/darwin/xdg/configFile/default.nix)

Maybe something more Lua-withPackages-y might be the better way to go, but I think that would be more about principles than concision.

**Comment by DavSanchez** [comment]
> @khaneliman I have SbarLua building with Nix and have integrated it into my setup. The packaging of the shared library isn't so bad:
> 
> https://github.com/shajra/shajra-provisioning/blob/main/build/nixpkgs/packages/sketchybar-lua/default.nix

Hey @shajra! Did you find any problems getting the built shared lib to work? Just trying to `require` this library but I'm finding SIGSEGVs, not sure how to debug this. Via dumb `print` statements, I got just where `require("sketchybar")` gets run, but not after:

```
'./sketchybarrc' terminated by signal SIGSEGV (Address boundary error)
```

Thanks!

**Comment by shajra** [comment]
> Hey @shajra! Did you find any problems getting the built shared lib to work? Just trying to `require` this library but I'm finding SIGSEGVs, not sure how to debug this. Via dumb `print` statements, I got just where `require("sketchybar")` gets run, but not after:
> 
> ```
> './sketchybarrc' terminated by signal SIGSEGV (Address boundary error)
> ```

Hi @DavSanchez, I'm still using SbarLua on my Mac with the same shajra/shajra-provisioning project referenced above.  So all the code in there works.

What I linked above is part of the project that builds this project.  But to integrate it into my configuration, I do that [here](https://github.com/shajra/shajra-provisioning/blob/main/home/modules/base/gui/darwin/xdg/configFile/default.nix).

Have a look around.  I know Nix can appear foreign at first glance, but this part of the code uses it more as a templating language.  I don't think I do anything that is out of the ordinary.  It's just automating with Nix the way @FelixKratz documents setting things up.

**Comment by DavSanchez** [comment]
Yes I tried to keep it simple and just use it directly with home-manager options, but not sure where's failing to work as expected. I just put the `.so` and the configs in the [default locations, with home-manager](https://github.com/DavSanchez/nix-dotfiles/blob/10e4948b72ab0583c27278c9967b02e3e1f7c710/home-manager/home-mbp.nix#L99-L120) and them attempt to load them :/

For the moment, I'm actually using the very same configs in use by FelixKratz's dotfiles... As for the SBarLua package itself, before I was packaging it in a slightly different way and was not working, tried yours but having the same errors.

Just tried copying the whole derivation output (`liblua.a`, `sketchybar.so` and the directory `sketchybar.so.dSYM`, I assume your output is the same) to `~/.local/share/sketchybar_lua`) and letting [this module](https://github.com/DavSanchez/nix-dotfiles/blob/10e4948b72ab0583c27278c9967b02e3e1f7c710/home-manager/darwin/sketchybar/helpers/init.lua#L2) set the lookup dirs to no avail.

**Comment by khaneliman** [comment]
If it helps, at all. I create a package for it like this: https://github.com/khaneliman/khanelinix/blob/main/packages/sbarlua/default.nix and just reference it in my rc like this https://github.com/khaneliman/khanelinix/blob/8919868593d213f7980b267c4567cede627db603/modules/home/programs/graphical/bars/sketchybar/default.nix#L45-L63

**Comment by DavSanchez** [comment]
I had considered the fact that a different Lua version than the one compiling the `.so` was running the script but didn't know this `makeFlags` existed. Thanks! I'll try this.

EDIT: Same error when following @khaneliman 's [approach](https://github.com/DavSanchez/nix-dotfiles/blob/a2ac83507b56b2f3ec872f4efd1124a25f09b768/home-manager/home-mbp.nix#L114). Not sure what's wrong 🤔 

```console
$ ./sketchybarrc
(cd event_providers && make) >/dev/null
(cd menus && make) >/dev/null
Attempting to require sketchybar...
fish: Job 1, './sketchybarrc' terminated by signal SIGSEGV (Address boundary error)
```

**Comment by Lalit64** [comment]
@khaneliman can you checkout my pull request #45
